webapp/platform/client/src/websocket.ts

```diff
@@ -358,8 +395,12 @@ export default class WebSocketClient {
     close() {
         this.connectFailCount = 0;
         this.responseSequence = 1;
+        if (this.reconnectTimeout) {
+            clearTimeout(this.reconnectTimeout);
+            this.reconnectTimeout = null;
+        }
```

Comment:
A close because of an error won't also call this `close` function? won't this break reconnects in general?

Reply:
I'm not following your question, maybe I'm misunderstanding something about the code here 😅 .

The `onerror` callback does not call `close()`. This `close()` function, if I understand correctly, is meant to completely close the websocket client and kill all connections. We shouldn't be trying to reconnect if this function is called.

This is different from the `onclose` callback that _does_ trigger reconnects.